### PR TITLE
Script to update neqo version

### PR DESCRIPTION
**Pull request by ddragana**
Some time ago I wrote a script to update neqo version. I will just post the PR as a draft, I do not want to add it to repo.

### CODE REVIEW

File: update_version [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#! /bin/bash
+
```
Comment (martinthomson) [inline update_version:2]:
```suggestion

cd "$(dirname "$0")"
set -e

[[ -n "$1" ]] || ! echo "Usage: $0 <version>"
```

File: update_version [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+for entry in `ls ./*/Cargo.toml`; do
```
Comment (martinthomson) [inline update_version:3]:
```suggestion
for entry in `find . -mindepth 2 -name Cargo.toml -print`; do
```

**Review by martinthomson — COMMENTED** [review]
I don't see why you wouldn't put this in-tree.  It's useful to have.

That said, I would just put the new version number in as "$1" and use that.  This might work better though:

```sh
find . -mindepth 2 -name Cargo.toml -exec \
  sed -i~ -e '0,/^version =/s/^version =.*/version = "'"$1"'"/' {} \+
```

File: update_version [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+cd "$(dirname "$0")"
+set -e
+
+[[ -n "$1" ]] || { echo "Usage: $0 <version>"; exit 1; }
+
+for entry in `find . -mindepth 2 -name Cargo.toml -print`; do
+    echo $entry
+    line=$(grep -n -m1 "version" $entry | cut -d: -f2)
+    CURRENT=$(echo ${line} | awk -F'"' '{print $2}')
+    sed -i "" "s/$CURRENT/$1/g" $entry
```
Comment (martinthomson) [inline update_version:12]: I think that you need a -e here (and the `-i ""` might fail on mac, I can only remember that `-i~` works everywhere.

Comment (ddragana) [reply update_version:12]: This works om my mac.

Comment (mb) [reply update_version:12]: -i "" fails on linux for me. I suggest removing the space between `-i` and `""` changing it to

```
sed -i"" "s/$CURRENT/$1/g" $entry
``` 

Which only doesn't works on MacOS up to 10.8[[1]](https://stackoverflow.com/a/4247319), which had end of support around September 2016[[2]](https://en.wikipedia.org/wiki/MacOS_version_history#Releases). So it works on all platforms I think we care about.

Comment (martinthomson) [reply update_version:12]: OK, good.  Note that `sed -i""` is exactly equivalent to just `sed -i`.

Comment (mb) [reply update_version:12]: Yes, on linux it is equivalent. On macos only `sed -i` doesn't work if I understand it correctly.

**Comment by larseggert** [comment]
I pushed a version that addresses the issues.

**Review by mb — APPROVED** [review]
Thanks. Looks good and works for me :+1:

**Pull request merged**